### PR TITLE
feat: add all apple family support

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> add support for all apple family devices

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/util/index.js [hunk 1]
```diff
@@ -1112,6 +1112,14 @@ const isOAuthSupported = (destination, destHandler) => {
   );
 };
 
+function isAppleFamily(platform) {
+  return (
+    platform.toLowerCase() === "ios" ||
+    platform.toLowerCase() === "watchos" ||
+    platform.toLowerCase() === "ipados" ||
+    platform.toLowerCase() === "tvos"
+  );
```
Comment (saikumarrs) [inline v0/util/index.js:1121]: Helps to easily extend in the future.
```suggestion
  const appleOSNames = ["ios", "watchos", "ipados", "tvos"];
  return appleOSNames.includes(platform.toLowerCase());
```

File: v0/destinations/clevertap/transform.js [hunk 2]
```diff
@@ -178,7 +178,7 @@ const responseBuilderSimple = (message, category, destination) => {
       if (
         get(message, "anonymousId") &&
         deviceToken &&
-        ["ios", "android"].includes(deviceOS)
+        ["ios", "watchos", "ipados", "tvos", "android"].includes(deviceOS)
```
Comment (saikumarrs) [inline v0/destinations/clevertap/transform.js:181]:
```suggestion
        (deviceOS === "android" || isAppleFamily(deviceOS))
```

**Review by saikumarrs — CHANGES_REQUESTED** [review]
Add test cases for all the destinations and souces

File: v0/util/index.js (hunk 1 above)
Comment (akashrpo) [reply v0/util/index.js]: Resolved

File: v0/destinations/clevertap/transform.js (hunk 2 above)
Comment (akashrpo) [reply v0/destinations/clevertap/transform.js]: Resolved

File: v0/util/index.js (hunk 1 above)
Comment (akashrpo) [reply v0/util/index.js]: Refactored

File: v0/destinations/clevertap/transform.js (hunk 2 above)
Comment (akashrpo) [reply v0/destinations/clevertap/transform.js]: Refactored

**Review by saikumarrs — APPROVED** [review]

**Review by desusai7 — APPROVED** [review]

**Pull request merged**